src/auth.rs

```diff
@@ -127,25 +129,44 @@ where
     }
 }
 
+trait RoleBounds<Role>: DynClone + Send + Sync {
+    fn contains(&self, role: &Role) -> bool;
+    fn is_range_full(&self) -> bool;
+}
+
+impl<T, Role> RoleBounds<Role> for T
+where
+    Role: PartialOrd + PartialEq + Clone + Send + Sync + 'static,
+    T: RangeBounds<Role> + Clone + Send + Sync,
+{
+    fn contains(&self, role: &Role) -> bool {
+        RangeBounds::contains(self, role)
+    }
+
+    fn is_range_full(&self) -> bool {
```

Comment:
I'm not a fan of this function, but didn't find a better way to compare `self` with `RangeFull`.